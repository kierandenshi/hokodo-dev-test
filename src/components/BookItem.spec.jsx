import React from 'react';
import { shallow } from 'enzyme';
import Component from './BookItem';

const defaultProps = {
  book: {
    id: 'book1',
    name: 'Really Good Book',
    author: {
      name: 'Mr. John Doe',
      id: 'author1',
    },
    cover: 'http://coverimage.url',
  },
};

const getWrapper = (testProps = {}) => {
  const props = { ...defaultProps, ...testProps };
  return shallow(<Component {...props} />);
};

describe('Component', () => {
  it('should match snapshot', () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});
