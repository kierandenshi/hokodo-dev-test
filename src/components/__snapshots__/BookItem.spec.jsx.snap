// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should match snapshot 1`] = `
<styled.div>
  <BookCover
    imageUrl="http://coverimage.url"
    size="thumbnail"
  />
  <div
    className="details"
  >
    <div
      className="title"
    >
      Really Good Book
    </div>
    <div
      className="author"
    >
      Mr. John Doe
    </div>
    <Styled(Link)
      to="/book1"
    >
      More information
    </Styled(Link)>
  </div>
</styled.div>
`;
